turbinia/workers/fsstat.py

```diff
@@ -38,18 +38,24 @@ def run(self, evidence, result):
         TurbiniaTaskResult object.
     """
     fsstat_output = os.path.join(self.output_dir, 'fsstat.txt')
+
     # Since fsstat does not support XFS, we won't run it when we know the
     # partition is XFS.
-    if evidence.path_spec.type_indicator == 'XFS':
-      message = 'Not running fsstat since partition is XFS'
-      result.log(message)
-      result.close(self, success=True, status=message)
-    else:
-      output_evidence = ReportText(source_path=fsstat_output)
-      cmd = ['sudo', 'fsstat', evidence.device_path]
-      result.log('Running fsstat as [{0!s}]'.format(cmd))
-      self.execute(
-          cmd, result, stdout_file=fsstat_output,
-          new_evidence=[output_evidence], close=True)
+
+    # Note: an evidence object that was serialized will not have a path_spec
+    # because it is removed prior to JSON serialization in evidence.py:243
```

Comment:
Cool, thanks for the fix @jleaniz !  I think this is a good safe fix for the short term, but I think it means that we will fail on XFS partitions since the evidence will always be serialized when it's sent back to the server.    @dfjxs or @jleaniz what do you think a good longer term fix for this would be -- Should we save this value to a new/separate property of the Partition Evidence type when it gets created?  Or maybe there is a more selective property of the path_spec that we could unset to make it serializeable again?

Reply:
Yeah, we could also just keep the type_indicator as an attribute since it's just a string ,instead of the whole path_spec object. I'm not sure we need to serialize path_spec. It seems like right now we don't do that anyway.

Reply:
Sure, saving it as a separate attribute sounds good for now since it looks like we don't need anything else from the path_spec. Should we do that as part of this PR, or just do that in a separate one?

Reply:
I'll add it to this PR

Reply:
I see it's added now, thanks!  Could you also set this attribute when the `DiskPartition` evidence gets created in `turbinia/workers/partitions.py`?

Reply:
Done. However, If i understand this correctly, the `path_spec` is generated for each task in the `_preprocess`function in `evidence.py` which is where I'm now grabbing the type_indicator from, and setting the new attribute at the same time. So even if it's set in the constructor in `partitions.py` it will get overwritten by the call to `_preprocess`. I believe this is also the reason why path_spec is not passed to the `DiskPartition` constructor in `partitions.py`

Reply:
Hrm, yeah, you're right, it does generate that in the pre-processor.  That happens on the worker just prior to executing the Task, so now I'm confused about how the serialization could be the original cause of this.   Is there maybe another issue that is causing `GetPathSpecByLocation()` to return None?